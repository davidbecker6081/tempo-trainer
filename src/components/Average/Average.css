.average {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 1rem;
  padding: 0 15px;
  width: 100%;
  background-color: #252525;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 2px #25252563;
  color: #FFFFFF;
}

.average h4 {
  font-size: 1.2em;
  font-weight: 300;
}

.average p {
  font-size: 2em;
  font-weight: 700;
}

.average p,
.average h4 {
  margin: 15px 0;
}
