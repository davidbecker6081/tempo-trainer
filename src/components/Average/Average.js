import React from 'react';
import PropTypes from 'prop-types';
import './Average.css';

const Average = ({ channelSet, dataHelper }) => {
  const average = dataHelper.calculateAverage(channelSet);
  const averageCleaner = {
    power: Math.round(average),
    heartRate: Math.round(average),
    cadence: Math.round(average),
    temperature: Math.round(average * 10) / 10,
    elevation: Math.round(average),
    speed: Math.round(average * 100) / 100,
  };

  return (
    <article>
      <h4>{channelSet.toUpperCase()}</h4>
      <p>{averageCleaner[channelSet]}</p>
    </article>
  );
};

export default Average;

Average.propTypes = {
  channelSet: PropTypes.string.isRequired,
  dataHelper: PropTypes.shape({
    data: PropTypes.shape({}),
    channels: PropTypes.arrayOf(PropTypes.string),
    GPSCoords: PropTypes.arrayOf(PropTypes.shape({})),
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};
