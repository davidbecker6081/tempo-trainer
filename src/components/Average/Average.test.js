import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Average from './Average';
import DataHelper from '../DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('Average', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      channelSet: '',
      dataHelper: {},
    };
  });

  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
