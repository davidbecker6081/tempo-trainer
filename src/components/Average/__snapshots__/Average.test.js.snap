// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average channelSet is defined should render correclty 1`] = `undefined`;

exports[`Average dataHelper is defined should render correclty 1`] = `undefined`;
